### DISCUSSION TITLE
How can I reply to someone?

### DISCUSSION THREAD

**Discussion by Sieun-Ju** [redouane59/twittered#321] [Q&A]
This is my first time making twitter bot using API. so forgive me if I don't know much.
I was wondering if I can make my bot to reply someone who mentions me.
Like if some mention "@mybotid HI" and maybe my bot can reply to that automatically.

**Comment by redouane59** [comment]
Hey, you have to use the stream endpoint and set up rule to trigger something when someone mention you.
The method names are `startFilteredStream` to start the stream, `addFilteredStreamRule` to add a new rule.
You will find all other details directly on the twitter API documentation.

**Reply by Sieun-Ju** [reply]
Thx for response!
I read the API documentation and method, you answered.
So I tried few things, but I don't think I'm getting it...

`String	rule = "hi to:mybotid"; 
List<StreamRule> rules = twitterClient.retrieveFilteredStreamRules();
twitterClient.addFilteredStreamRule(rule, "say hi me");
Future<Response> future = twitterClient.startFilteredStream(System.out::println);
try {
	 future.get(5, TimeUnit.SECONDS);
		      
} catch (TimeoutException exc) {
}	
StreamMeta streamMeta = twitterClient.deleteFilteredStreamRule(rule);
System.out.print("End");`



I still don't know how to get filtered tweets or tweet list. 
Sorry to bother you.

**Reply by redouane59** [reply]
here is a sample for answers : 

```
public void start() {

    try {
      System.out.println("start streaming... ");
      this.streamResponse = TWITTER_CLIENT.startFilteredStream(new IAPIEventListener() {
        @Override
        public void onTweetStreamed(final Tweet tweet) {
          try {
              TweetParameters tweetParameters = TweetParameters.builder()
                                                               .text("Hi")
                                                               .reply(Reply.builder().inReplyToTweetId(tweet.getId()).build())
                                                               .build();
              TWITTER_CLIENT.postTweet(tweetParameters);
            }
          } catch (Exception e) {
            System.err.println("error streaming tweet : " + e.getMessage());
          }
        }

      // ...
  }
```
